### ISSUE TITLE
✨ Set checkOrigin: false as default (or create a template with this preset)

### ISSUE THREAD

**Issue by khromov** [SvelteLab/SvelteLab#229]
### Describe the problem

Based on https://github.com/SvelteLab/SvelteLab/issues/197 it looks like Form Actions aren't working on SvelteLab out of the box. It's frustrating because that's a core SvelteKit feature.

### Describe the solution you would like

I'm not an expert on WebContainers, perhaps setting `checkOrigin: false` by default wouldn't be so bad, since they are isolated in the browser?

### Describe alternatives you have considered

* Adding a template for it
* Having a command form the command palette to automatically add this

### Additional context

Related Discord thread:
https://discord.com/channels/457912077277855764/1116492773353340938

**Comment by paoloricciuti** [comment]
Yeah I thought of adding a command to "enable form"...the problem is that is not completely straightforward: it needs to add the value `kit.csrf = false` to the `svelte.config.js` but you can't just replace the existing `svelte.config.js` in case it was already modified, you can't even use regex because while the default is to write `export default const config` someone might have change that, or might have added another object to the file. So I would need to statically analize the code, found the default export, add the property `kit.csrf` if it doesn't exist and set it to false.

It's certainly doable but not straightforward, especially not without bringing a huge library for code parsing.

But we are definitely thinking of this and if you have ideas on how to implement this simply I would love to hear them.

**Comment by khromov** [comment]
Could it be an option to add `checkOrigin: true` by default (already an improvement because it helps people with a clue) and then run a regex on `svelte.config.js`, something like `js.replace(/"checkOrigin":\s*false/g, '"checkOrigin": true');`. It won't work if someone modified the config heavily but it'd work out of the box and probably most people don't even customize svelte.config.js (perhaps you have stats on that?)

**Comment by paoloricciuti** [comment]
> Could it be an option to add `checkOrigin: true` by default (already an improvement because it helps people with a clue) and then run a regex on `svelte.config.js`, something like `js.replace(/"checkOrigin":\s*false/g, '"checkOrigin": true');`. It won't work if someone modified the config heavily but it'd work out of the box and probably most people don't even customize svelte.config.js (perhaps you have stats on that?)

I don't think most people customize it that much but I'm always skeptic to assume that 😄 

Your solution could be good tho because I don't think there are other checkOrigin fields of the config so even if they do weird things it should still catch it up. The issue of someone opening a GitHub repo without checkOrigin: true remains tho.

Also also I have the feeling that even having a command would not be enaugh for people because you'd have to know that that is an option. I have to check if there's any way we can detect a failing form submission so that we can show a guide in a toast message.

**Comment by manuel3108** [comment]
> Yeah I thought of adding a command to "enable form"...the problem is that is not completely straightforward: it needs to add the value kit.csrf.checkOrigin = false to the svelte.config.js but you can't just replace the existing svelte.config.js in case it was already modified, you can't even use regex because while the default is to write export default const config someone might have change that, or might have added another object to the file. So I would need to statically analize the code, found the default export, add the property kit.csrf.checkOrigin if it doesn't exist and set it to false.

We pretty much have that functionality present in `svelte-add`. We could think of creating a small adder, that does exactly this. Still the detection would need to be necessary on your side.
